server/models/User.ts

```diff
@@ -616,6 +635,127 @@ class User extends ParanoidModel<
       ],
     });
 
+  /**
+   * Determines whether the user has a higher access level to a document.
+   *
+   * @param {Object} params Input parameters.
+   * @param {string} params.documentId The document to check.
+   * @param {DocumentPermission} params.permission The base permission to compare against.
+   * @param {string} params.skipMembershipId The membership to skip when comparing the existing permissions.
+   * @returns {boolean} Whether the user has a higher access level
+   */
+  public async hasHigherDocumentPermission({
+    documentId,
+    permission,
+    skipMembershipId,
+  }: {
+    documentId: string;
+    permission: DocumentPermission;
+    skipMembershipId?: string;
+  }) {
+    const existingPermission = await this.getDocumentPermission(
+      documentId,
+      skipMembershipId
+    );
+
+    if (!existingPermission) {
+      return false;
+    }
+
+    return (
+      DocumentPermissionPriority[existingPermission] >=
+      DocumentPermissionPriority[permission]
+    );
+  }
+
+  /**
+   * Returns the user's highest access level to a document.
+   *
+   * @param {string} documentId The document to check.
+   * @param {string} skipMembershipId The membership to skip when comparing the existing permissions.
+   * @returns {DocumentPermission | undefined} Highest access level, if it exists.
+   */
+  public async getDocumentPermission(
```

Comment:
```suggestion
  private async getDocumentPermission(
```

I don't love this being on `user`, but if it must we can make it private

Reply:
yes, I tried moving just this method to the existing "policies" file (renamed to permissions), but it led to circular imports with `User`, so I had to move both methods out (figured a separate `getDocumentPermission` would be beneficial in future).